drivers/builtin/src/entropy_poll.c

```diff
@@ -15,24 +15,49 @@
 #include "mbedtls/entropy.h"
 #include "entropy_poll.h"
 #include "mbedtls/error_common.h"
+#include <psa/crypto_driver_random.h>
+
+/* In principle, we could support both a built-in source and a custom
+ * source. However, it isn't a common need. So for now the two
+ * callback functions have the same name and there can only be one. */
+#if defined(MBEDTLS_PSA_BUILTIN_GET_ENTROPY) || defined(MBEDTLS_PSA_DRIVER_GET_ENTROPY)
+
+/* We currently only support a single "true" entropy source (other than the
+ * "fake" source which is the NV seed). It can be either the built-in one
+ * or a user-provided callback. */
+#if defined(MBEDTLS_PSA_DRIVER_GET_ENTROPY) && defined(MBEDTLS_PSA_BUILTIN_GET_ENTROPY)
+#error "MBEDTLS_PSA_DRIVER_GET_ENTROPY and MBEDTLS_PSA_BUILTIN_GET_ENTROPY " \
+    "are currently incompatible."
+#endif
 
 int mbedtls_entropy_poll_platform(void *data, unsigned char *output, size_t len, size_t *olen)
 {
     int ret;
-    size_t entropy_content = 0;
+    size_t estimate_bits = 0;
     (void) data;
 
-    ret = mbedtls_platform_get_entropy(output, len, olen, &entropy_content);
+    /* Historically, in PolarSSL and Mbed TLS, the entropy callback provided
+     * full-entropy output, and reported how many bytes in the output buffer
+     * had useful data. Reporting the length was not very useful because
+     * we get the same amount of entropy by processing the whole output bufer,
+     * processing the whole buffer barely costs more CPU time since the buffer
+     * is small, and most entropy sources just fill the whole buffer anyway.
+     * So since TF-PSA-Crypto 1.0, we process the whole buffer.
+     */
+    *olen = len;
+
+    ret = mbedtls_platform_get_entropy(0, &estimate_bits, output, len);
```

Comment:
```suggestion
    ret = mbedtls_platform_get_entropy(PSA_DRIVER_GET_ENTROPY_FLAGS_NONE, &estimate_bits, output, len);
```
We can now take advantage of the newly defined `PSA_DRIVER_GET_ENTROPY_FLAGS_NONE` ;)